pkgs/misc/emulators/yuzu-mainline/default.nix

```diff
@@ -0,0 +1,41 @@
+{ stdenv, mkDerivation, fetchgit
+, cmake, libressl, SDL2, qtbase, python2, alsaLib
+, useVulkan ? true, vulkan-loader, vulkan-headers
+}:
+
+mkDerivation rec {
+  pname = "yuzu-mainline";
+  version = "unstable-2020-04-13";
+
+  # Submodules
```

Comment:
Why not use `fetchFromGithub`?

Reply:
> Why not use fetchFromGithub with fetchSubmodules?

I didn't know that made a difference, as this already fetches the submodules. Is `fetchFromGithub` preferred?

> The libressl issue seems to have been fixed by yuzu-emu/yuzu#3675. That also allows us to use more dependencies from nixpkgs so we should include that.

I wanted to check that out as well, but Linux compilation on master is currently so I couldn't really play with it. the rev this is using doesn't have the PR merged yet.

Reply:
> I didn't know that made a difference, as this already fetches the submodules. Is `fetchFromGithub` preferred?

It results in slightly cleaner expression (but uses `fetchgit` internally)-


> I wanted to check that out as well, but Linux compilation on master is currently so I couldn't really play with it. the rev this is using doesn't have the PR merged yet.

Is there an upstream issue for that?

Reply:
> It results in slightly cleaner expression (but uses fetchgit internally)-

Alright, will switch over tomorrow (No pun intended).

> Is there an upstream issue for that?

[Yes](Is there an upstream issue for that?), should be a quick and easy fix.

Reply:
I originally used `fetchgit` in this way because it was used in the `citra` package, and it seemed to work without the hashing issue.

You're right though, fetchFromGithub is slightly cleaner and I prefer it.